### PR TITLE
fix: $destroy and createRoot are no more

### PR DESCRIPTION
**Pull request by yanick**

### CODE REVIEW

**Review by mcous — APPROVED** [review]
Feel free to ping me once this is in and I'll rebase #325 accordingly

**Comment by mcous** [comment]
I added the `mount` / `unmount` changes to #325 before you made this PR, which will at least unblock the majority of testing. I'm starting to think that either:

- `rerender` won't be supported in Svelte 5
    - Instead, the complexity is pushed to the user, who will need to do something like
      ```js
      // comp.test.svelte.js
      const props = $state({ name: 'hello' })
      render(Comp, { props })
      // ...
      act(() => (props.name = 'goodbye'))
      ```
    - This is how `solid-testing-library` operates
- We'll need to introduce a build step into this library so we can use `$state` ourselves
    - I actually don't know if this will work

Either path feels like a bit of a bummer!

**Comment by yanick** [comment]
>     * `rerender` won't be supported in Svelte 5
>       
>       * Instead, the complexity is pushed to the user, who will need to do something like
>         ```js
>         // comp.test.svelte.js
>         const props = $state({ name: 'hello' })
>         render(Comp, { props })
>         // ...
>         act(() => (props.name = 'goodbye'))
>         ```
>       * This is how `solid-testing-library` operates

Yup, that's what I think as well. I'm trying to change the rerender test to use that ^^^ pattern, but for some reason it doesn't react to `props.name = 'goodbye'`. But I used a waitFor, not an act, so I'll try that.

> 
>     * We'll need to introduce a build step into this library so we can use `$state` ourselves

I *think* we just need to change the name of the test file to be `test.svelte.js` for the compiler to do the right thing. I'll say this with more conviction once the test is passing. :-)

Right now my battle plan is

1. find out the Way That Works with Svelte 5.
2. see how that can fit with the previous versions of svelte. Worst case we might have to split things and have `import {} from '@testing-library/svelte` for the latest supported version and `import {} from '@testing-library/svelte/svelteX` for `X` that are legacy versions and upcoming versions. 
3. Depending on (2), organize the tests to cover the maximum without going insane.

So yeah, weeeeee, supporting several major versions is fun, fun, fun. :zany_face: 


>       
>       * I actually don't know if this will work
> 
> 
> Either path feels like a bit of a bummer!

**Comment by yanick** [comment]
@mcous This is not the patch we deserve, but probably the patch we need. I'm using the `svelte/legacy` API of svelte 5 as I could not make the runes work in the testing context yet. I've also skipped 2 tests for svelte 5 and happy-dom because they are failing in a weird way.

File: src/svelte5.js
Comment (mcous) [inline src/svelte5.js:1]: This feels like a lot of duplication for little payoff. If you want to have separate entry points, would it be better to re-export `pure.js`'s exports and only override the public methods that changed?

**Review by mcous — CHANGES_REQUESTED** [review]
Requesting changes because this breaks type exports in `package.json` for the existing library.

Additionally, the separate entry points for v4 vs v5 seems a little unnecessary from a technical standpoint, because the public APIs remain unchanged. What's the value of it this approach over what's currently in #325, where `pure.js` continues to support both?

File: package.json
Comment (yanick) [reply package.json:8]: Doesn't the `types` key right below the exports take care of that, though?

File: src/__tests__/context.test.js
Comment (yanick) [reply src/__tests__/context.test.js:7]: > should we have a little __tests__/environment.ts helper file that exports IS_SVELTE_5, IS_HAPPY_DOM, IS_JSDOM, etc. to make these a little less repetitive?

Yeah, that would make a lot of sense.

File: src/pure.js
Comment (yanick) [reply src/pure.js:14]: As mentioned in the previous comment, it's to deal with `import from 'svelte/legacy'`. We could probably dig out darker magic to do it another way, but I do feel there will be other things in Svelte 5 that will be easier if we just have different files implementing the same(ish) interface rather than having a file that tries to satisfy all versions at once.

File: src/svelte5.js
Comment (yanick) [reply src/svelte5.js:1]: My development steps are

1. Make it
2. Make it work
3. Make it pretty

This PR is at stage (2). all the functions that aren't different in `svelte5.js` should really be imported from `pure.js`. But first I was interested to see how deep the hole had to be. And then backfill and tidy up the garden again.

File: package.json
Comment (mcous) [reply package.json:8]: Unfortunately, no. If [TS sees `"exports"`](https://www.typescriptlang.org/docs/handbook/modules/reference.html#packagejson-exports) in `moduleResolute: nodenext` or `bundler` and there's no `exports[].types` it'll get sad

See #228

File: src/pure.js
Comment (mcous) [reply src/pure.js:14]: Doh, yeah duh, my bad, that makes total sense.

---
Aside: this makes me think dropping 3.0 in the `next` branch (in a separate PR to be responsible about it) will make things easier.

I'm not totally sure what to do about collecting multiple breaking changes without a prerelease version. Maybe we should switch over to an `beta` branch for breaking changes? Or consider switching `next` to be a prerelease in `semantic-release`.

File: src/svelte5.js
Comment (mcous) [reply src/svelte5.js:1]: Sure thing, happy to defer to whatever approach you want to take. Just wanted to point out opportunities nip stuff in the bud LoC-wise

**Pull request merged**
